unstructured_ingest/v2/processes/connectors/discord.py

```diff
@@ -0,0 +1,169 @@
+import datetime as dt
+from dataclasses import dataclass
+from typing import Any, Generator, List, Optional, Set
+
+from pydantic import Field, Secret
+
+from unstructured_ingest.logger import logger
+from unstructured_ingest.utils.dep_check import requires_dependencies
+from unstructured_ingest.v2.interfaces import (
+    AccessConfig,
+    ConnectionConfig,
+    Downloader,
+    DownloaderConfig,
+    DownloadResponse,
+    FileData,
+    FileDataSourceMetadata,
+    Indexer,
+    IndexerConfig,
+    SourceIdentifiers,
+)
+from unstructured_ingest.v2.processes.connector_registry import SourceRegistryEntry
+
+CONNECTOR_TYPE = "discord"
+
+
+class DiscordAccessConfig(AccessConfig):
+    token: str = Field(description="Discord API token")
+
+
+class DiscordConnectionConfig(ConnectionConfig):
+    access_config: Secret[DiscordAccessConfig]
+    channels: Optional[List[str]] = Field(
+        default=None, description="List of Discord channel IDs to process"
+    )
+
+
+class DiscordIndexerConfig(IndexerConfig):
+    recursive: bool = Field(
+        default=False, description="Recursively process child channels and messages"
+    )
+
+
+@dataclass
+class DiscordIndexer(Indexer):
+    connection_config: DiscordConnectionConfig
+    index_config: DiscordIndexerConfig
+
+    @requires_dependencies(["discord"], extras="discord")
+    def get_client(self):
+        import discord
+
+        intents = discord.Intents.default()
+        intents.message_content = True
+        return discord.Client(intents=intents)
+
+    def run(self, **kwargs: Any) -> Generator[FileData, None, None]:
+        client = self.get_client()
+        processed_channels: Set[str] = set()
```

Comment:
`set` is also a native type in python now. 
```suggestion
        processed_channels: set[str] = set()
```